.devcontainer/postCreate.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Commands to run to configure the devcontainer after it starts (since they require files in the repo).
+sudo gem install bundle
+sudo gem install bundler -v 2.2.30
+sudo gem install nokogumbo scrypt sanitize
+sudo chown -R vscode:vscode /workspaces/canvas-lms/
+sudo chown -R vscode:vscode /var/lib/gems/2.7.0/
+bundle _2.2.30_ install --without pulsar
+yarn install --pure-lockfile
+for config in amazon_s3 delayed_jobs domain file_store outgoing_mail security external_migration dynamic_settings database; \
+          do cp -v config/$config.yml.example config/$config.yml; done
+sudo chown -R vscode:vscode /var/lib/gems/2.7.0/
+bundle _2.2.30_ update
+bundle exec rails canvas:compile_assets
+sudo chown -R vscode:vscode /var/run/postgresql/
+export PGHOST=localhost
+/usr/lib/postgresql/12/bin/initdb ~/postgresql-data/ -E utf8
+/usr/lib/postgresql/12/bin/pg_ctl -D ~/postgresql-data/ -l ~/postgresql-data/server.log start
+/usr/lib/postgresql/12/bin/createdb canvas_development
+# Set correct domain for port forwarding on Codespaces. npx package was developed by a GitHub employee, since you can't get the URL directly in the codespace yet.
+if [ ! -z ${CODESPACES+x} ]
+then
+    sed -i "s/localhost:3000/$(npx -y codespaces-port 3000 | cut -c 9-)/g" config/domain.yml
+fi
+touch /home/vscode/.setupDone
+# bundle exec rails db:initial_setup
```

Comment:
Why didn't we finish running this here? With Codespaces [Prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces) you could wind up with an almost instant environment. @jedieaston

Reply:
It asked for user input, and at the time I didn't look into how to automate that. I might pick it back up

Reply:
Perhaps you found it, but you can predefine those input values with environment variables:
```
[...]
bundle exec rails canvas:compile_assets_dev &&
CANVAS_LMS_STATS_COLLECTION=opt_out \
CANVAS_LMS_ACCOUNT_NAME=School \
CANVAS_LMS_ADMIN_EMAIL=test@school.edu \
CANVAS_LMS_ADMIN_PASSWORD=password12345 \
bundle exec rake db:initial_setup
```
And if you want to run this in Prebuild as @octosteve suggests (and me too), just make sure to run `postCreate.sh` with "**on**CreateCommand" and not "**post**CreateCommand" in the `devcontainer.json`

Reply:
https://github.com/instructure/canvas-lms/pull/2038/files#diff-c26edb4d2e5f940d49820200be62a92f37f0e2c40c85d237207989d5d6d0d64fR10-R13 For an example.

Reply:
I'm going to close this since @octosteve's is farther along. Thanks for the tip though!